### ISSUE TITLE
Overflow Error

### ISSUE THREAD

**Issue by maxlxl** [vberlier/nbtlib#12]
Hello, i'm trying to load the nbt data extracted from a .mca file.
I already got a working program to extract the nbt data und decompress it. But when i try to load them i get an OverflowError:
```
Traceback (most recent call last):
  File "main.py", line 27, in <module>
    chunk_nbt_data = nbtlib.File().parse(io.BytesIO(decompressed_data))
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 496, in parse
    self[name] = cls.get_tag(tag_id).parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 496, in parse
    self[name] = cls.get_tag(tag_id).parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 496, in parse
    self[name] = cls.get_tag(tag_id).parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 224, in parse
    return cls(read_numeric(cls.fmt, buff, byteorder))
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 218, in __new__
    if cls.range and int(self) not in cls.range:
OverflowError: Python int too large to convert to C ssize_t
```
I also tried to save the extracted nbt-bytes to a file, but i've got the same result:
```
Traceback (most recent call last):
  File "main.py", line 30, in <module>
    chunk_nbt_data = nbtlib.load("tmp.nbt")
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/nbt.py", line 37, in load
    return File.from_buffer(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/nbt.py", line 96, in from_buffer
    self = cls.parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 496, in parse
    self[name] = cls.get_tag(tag_id).parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 496, in parse
    self[name] = cls.get_tag(tag_id).parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 496, in parse
    self[name] = cls.get_tag(tag_id).parse(buff, byteorder)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 224, in parse
    return cls(read_numeric(cls.fmt, buff, byteorder))
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/nbtlib/tag.py", line 218, in __new__
    if cls.range and int(self) not in cls.range:
OverflowError: Python int too large to convert to C ssize_t
```

I already searched but i couldn't find any simple solution by myself to fix this. If you need any additional information, let me know and thank you for your help!

**Comment by vberlier** [comment]
Hmm that's really weird can you give me a link to the file you're using, the "tmp.nbt" one?

**Comment by maxlxl** [comment]
Yeah sure, here you go: [tmp.nbt](https://cloud.maxlxl.de/index.php/s/ib6SHYm98Xpmzy4)
It's in binary and already the decompressed form of the original zlib data included in the .mca file for just a single chunk.

**Comment by vberlier** [comment]
I haven't been able to reproduce the issue. For me, the file loads just fine:

```py
>>> import nbtlib
>>> chunk_nbt_data = nbtlib.load('tmp.nbt')
>>> chunk_nbt_data.root.keys()
dict_keys(['Level', 'DataVersion'])
>>> chunk_nbt_data.root['DataVersion']
Int(1343)
>>> len(chunk_nbt_data.root['Level']['Sections'])
6
```

I'm running python 3.7.2 on linux and I'm using `nbtlib` version 1.4.1. Can you tell me more about your environment?

**Comment by vberlier** [comment]
Just tested it on Windows this time and again I managed to load the file just fine. Maybe try upgrading `nbtlib` to the latest version? I honestly have no idea.

**Comment by maxlxl** [comment]
Update: I've installed python3.7.2 on my macos and finally it works. I still don't know exactly, what the cause of the problem was, but for me it's enough by now. You can decide if you want to close this issue or not. (Maybe update the README.md with the new required python version for osx)

**Comment by vberlier** [comment]
Alright glad you managed to get it working. I'm going to try and setup a Travis job on osx to be sure that the problem can be replicated before changing the required python version for osx. Then I'll close the issue.

**Comment by vberlier** [comment]
Seems like I managed to replicate the problem https://travis-ci.com/vberlier/nbtlib/jobs/175466144.

**Comment by vberlier** [comment]
Alright this is fixed in 6c1d3fda2c1980c8122a59f9680bd6da9911e384.

**Comment by vberlier** [comment]
Just published version 1.4.2 which contains the fix for python 3.6 on macOS. I'm closing the issue but you're welcome to try it out.

**Comment by maxlxl** [comment]
Thank you very much for the detailled explanation.
I tried it again with version 1.4.2 and I can confirm that it works with Python3.6.0.

Keep up your good work! :)
